File: NationalSecurityCommunity/app/src/main/java/com/national/security/community/injection/component/ActivityComponent.java
package com.national.security.community.injection.component;

import com.national.security.community.injection.module.ActivityModule;
import com.national.security.community.injection.scope.ActivityScope;
import com.national.security.community.ui.main.MainActivity;
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    RxAppCompatActivity getActivity();

    void inject(MainActivity mainActivity);

}

File: app/src/main/java/com/jmd_news_kotlin/di/component/ActivityComponent.kt
package com.jmd_news_kotlin.di.component

import android.app.Activity
import com.jmd_news_kotlin.MainActivity
import com.jmd_news_kotlin.di.module.ActivityModule
import com.jmd_news_kotlin.di.scope.ActivityScope
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

    fun getActivity(): Activity
    fun inject(mainActivity: MainActivity)
}